docker/context/context.py

```diff
@@ -0,0 +1,190 @@
+        self.endpoints = {
+            "docker": {
+                "Host": utils.parse_host(None, IS_WINDOWS_PLATFORM),
+                "SkipTLSVerify": False
+            }
+        } if not endpoints else endpoints
+        self.tls_cfg = {}
+        self.meta_path = "IN MEMORY"
+        self.tls_path = "IN MEMORY"
+
+    def set_endpoint(
+            self, name, host=None, tls_cfg=None, skip_tls_verify=False,
+            def_namespace=None):
+        if not name:
+            return
+        self.endpoints[name] = {
+            "Host": utils.parse_host(host, IS_WINDOWS_PLATFORM),
+            "SkipTLSVerify": skip_tls_verify
+        }
+        if def_namespace:
+            self.endpoints[name]["DefaultNamespace"] = def_namespace
+
+        if tls_cfg:
+            self.tls_cfg[name] = tls_cfg
+
+    def inspect(self):
+        return self.__call__()
+
+    @classmethod
+    def load_context(cls, name):
+        name, orchestrator, endpoints = Context.load_meta(name)
+        if name:
+            instance = cls(name, orchestrator, endpoints)
+            instance.load_certs()
+            instance.meta_path = get_meta_dir(name)
+            return instance
+        return None
+
+    @classmethod
+    def load_meta(cls, name):
+        # load metadata
+        metadata = {}
+        meta_file = get_meta_file(name)
+        if os.path.isfile(meta_file):
+            with open(meta_file) as f:
+                try:
+                    with open(meta_file) as f:
+                        metadata = json.load(f)
+                    for k, v in metadata["Endpoints"].items():
+                        metadata["Endpoints"][k]["SkipTLSVerify"] = bool(
+                            v["SkipTLSVerify"])
+                except (IOError, KeyError, ValueError) as e:
+                    # unknown format
+                    raise Exception("""Detected corrupted meta file for
+                        context {} : {}""".format(name, e))
+
+            return (
+                metadata["Name"], metadata["Metadata"]["StackOrchestrator"],
+                metadata["Endpoints"])
+        return None, None, None
+
+    def load_certs(self):
+        certs = {}
+        # set certs
+        tls_dir = get_tls_dir(self.name)
+        for endpoint in self.endpoints.keys():
+            # get extension
+            if not os.path.isdir(os.path.join(tls_dir, endpoint)):
+                continue
+
+            ca_cert = None
+            cert = None
+            key = None
+            for filename in os.listdir(os.path.join(tls_dir, endpoint)):
+                if filename.startswith("ca"):
+                    ca_cert = os.path.join(tls_dir, endpoint, filename)
+                elif filename.startswith("cert"):
+                    cert = os.path.join(tls_dir, endpoint, filename)
+                elif filename.startswith("key"):
+                    key = os.path.join(tls_dir, endpoint, filename)
+            if all([ca_cert, cert, key]):
```

Comment:
`else`?

What happens if a file is renamed not to start with `key`, `ca` or `cert`? How can I know?